tests/eso/test_online.py

```diff
@@ -0,0 +1,296 @@
+        for p in template.parameters
+    )
+    new_params = {
+        "TEL.GS1.ALPHA": "11:22:33.000",
+        "INS.DROT.MODE": "SKY",
+        "INS.ADC.MODE": "AUTO",
+    }
+    template = facility.update_template_params(ob, template, new_params)
+    # Check updated rotator mode is SKY
+    assert any(
+        p.get("name") == "INS.DROT.MODE" and p.get("value") == "SKY"
+        for p in template.parameters
+    )
+    facility.delete_template(ob, template)
+    # Need to refresh observation block
+    ob = facility.get_ob(ob.ob_id)
+    facility.delete_ob(ob)
+    # Need to refresh the container
+    folder = facility.get_container(folder.container_id)
+    facility.delete_container(folder)
+
+
+@pytest.mark.side_effect
+def test_save_absolute_time_constraints():
+    facility = EsoFacility()
+    folder = facility.create_folder(
+        ESO_TUTORIAL_CONTAINER_ID, "AEONlib.test_save_abs_time_con"
+    )
+    ob = facility.create_ob(folder, "AEONLIB.test_save_abs_time_con.ob")
+    abs_constraints = AbsoluteTimeConstraints(
+        constraints=[
+            AbsoluteTimeConstraint(
+                start=datetime.now(), end=datetime.now() + timedelta(days=30)
+            )
+        ]
+    )
+    new_abs_constraints = facility.save_absolute_time_constraints(ob, abs_constraints)
+    # Saving loses some precision on time, so just compare the date for this test
+    assert (
+        new_abs_constraints.constraints[0].start.date()
+        == abs_constraints.constraints[0].start.date()
+    )
+    assert (
+        new_abs_constraints.constraints[0].end.date()
+        == abs_constraints.constraints[0].end.date()
+    )
+    facility.delete_ob(ob)
+    # Need to refresh the container
+    folder = facility.get_container(folder.container_id)
+    facility.delete_container(folder)
+
+
+@pytest.mark.side_effect
+def test_save_sidereal_time_constraints():
+    facility = EsoFacility()
+    folder = facility.create_folder(
+        ESO_TUTORIAL_CONTAINER_ID, "AEONlib.test_save_sidereal_time_con"
+    )
+    ob = facility.create_ob(folder, "AEONLIB.test_save_sidereal_time_con.ob")
+    sidereal_constraints = SiderealTimeConstraints(
+        constraints=[
+            SiderealTimeConstraint(
+                start=time(2, 0),
+                end=time(12, 0),
+            )
+        ]
+    )
+    new_sidereal_constraints = facility.save_sidereal_time_constraints(
+        ob, sidereal_constraints
+    )
+    assert (
+        new_sidereal_constraints.constraints[0].start
+        == sidereal_constraints.constraints[0].start
+    )
+    assert (
+        new_sidereal_constraints.constraints[0].end
+        == sidereal_constraints.constraints[0].end
+    )
+    facility.delete_ob(ob)
+    # Need to refresh the container
+    folder = facility.get_container(folder.container_id)
+    facility.delete_container(folder)
+
+
+@pytest.mark.skip(reason="Errors with 'Ephemeris file must have at least two records.'")
```

Comment:
@jchate6 could not get this test to pass for the life of me. 
In fact, even using the p2 web ui I was not able to download an ephermis.txt from their tool and upload it successfully, I would get the same error from the web UI as this test gets. Maybe you could help me with it at some point?